core/src/main/java/org/opensearch/sql/catalog/model/Catalog.java

```diff
@@ -0,0 +1,27 @@
+/*
+ *
+ *  * Copyright OpenSearch Contributors
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.opensearch.sql.catalog.model;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.opensearch.sql.storage.StorageEngine;
+
+@Getter
+@RequiredArgsConstructor
+@EqualsAndHashCode
+public class Catalog {
```

Comment:
Thanks for adding this abstraction. Please think about if any previous added logic in Analyzer and other class can be moved here. For now it's just a data class without any behavior. I think it will show the power once we hide complexity behind it.

Reply:
`Currently the only use case is getStorageEngine and Table from it` which can be served with data class itself.

I see this could be extended with more behavior included. In the subsequent tasks, will try to use this abstraction more.